### DISCUSSION TITLE
Ability to remove follower

### DISCUSSION THREAD

**Discussion by vuedj** [community/community#18941] [Other Feature Feedback, Questions, & Ideas]
Github should allow the account holder the ability to **remove follower(s)** from their followers list.
I don’t want to see spam Github accounts following my Github profile.

Steps to reproduce:

1) Login to github
2) Click on “Your profile”
3) On this page, click on followers - you will see people who are following you - Github should provide the ability to remove follower(s) if the account holder wishes to do so.

Github - please action this - Instagram follows the same policy - the instagram user can remove follower(s) from their profile - similar kind of feature is needed here

**Comment by melvincwng** [comment]
@anykate, to remove a follower, I think the current workaround solution is to block that particular user.

This should remove the follower from your followers list.

Hope this helps

**Reply by vuedj** [reply]
That's precisely why I mentioned Instagram's way of handling it - Some users like me would like to remove the follower without blocking them.

Instagram has both options - either block them OR remove them from your followers list - it should be upto you to decide what approach you want to take.

Hence requesting that in github too, both options should be available - out of which "blocking" a user is already available - however, ability to remove the user without blocking them should also be available is what I want to say.

**Reply by vuedj** [reply]
Also, if I block a user, they will be added to the blocked user list -> accessible via Settings -> Access -> Moderation -> Blocked users and the list will continue to grow....I dont want to keep any trace of this account that I am removing in my settings

Secondly, I have to go to each user's profile and then block them....Instead if the "Remove follower" button appears on the listing page itself where I see all my followers, it becomes much easier to remove the ones that I dont want....

Hope this explanation helps!

**Reply by James-E-A** [reply]
@anykate just unblock them 0.01 seconds later; [they will stay unfollowed](https://github.com/orgs/community/discussions/18941#discussioncomment-2990976), and you won't have a long blocklist to keep track of that way

**Comment by jok1n9** [comment]
yeah i think block is the only option ...

**Comment by ZaidKhan144** [comment]
Curious, why would you want to remove a follower?

**Reply by An0n-00** [reply]
This is a good question. If you want to be a proffesional looking guy, you won't want to have bots or follow-me-back-accounts following you. This simply looks bad (in my opinion oc). 

For example:
![image](https://github.com/community/community/assets/142223573/9028273e-f1b1-4a22-97fb-ea01e929150b)
![image](https://github.com/community/community/assets/142223573/3ceee23a-caf0-4c62-b92f-c4091ffeb918)

Hell, I dont want some random russian dude following me on github (no offence against russian dudes oc 🙃 )

**Reply by killuaFreecs** [reply]
there must be a mistake followers and following are swapped 💯

**Reply by dreua** [reply]
Why do you think it is swapped? This person has most likely a bot/script to follow people more or less at random. I call this spam. The mistake is that GH is doing nothing to prevent this kind of abuse/spam. (And our mistake is to stay on this platform although it is getting increasingly enshittified.)

I guess some people (around 1 in 62) see that they got a new follower and instead of blocking it, they click follow back. Not a great ratio but still a significant amount of followers assuming that the following is free of charge.

**Comment by rilysh** [comment]
For now GitHub don't have any features releated to this and I don't think they'd ever going to do anything for this, as it's even considered not a feature to remove a follower.

But if you want to do that, block a user (they'll be removed from your follower list), and unblock the user after blocking, therefore it will provide the same thing that you've asked.

If you don't want that a specific user follows you, block or making profile private are only ways to go.

**Comment by p14** [comment]
I know a few people have mentioned not wanting to acquire a blocked list, but I believe the best option would be to block the user, then immediately unblock them. This will remove the follow, no notifications will be sent, and will effectively solve your issue. I know it would be nice to have something more straightforward (such as an unfollow button), but this is probably the best way to handle it on almost any social platform.

I hope this helps!

**Comment by ghost** [comment]
Hi all! @anykate @p14 @jok1n9 @Ruzie @ZaidKhan144 

#### 1. source-code
```
// @ domain: https://github.com/username?tab=following
var buttons = document.getElementsByClassName('btn btn-sm');

 for(var i = 0; i < buttons.length; i++)  
     buttons[i].click()
```

#### 2. notes
remover all follower in github 

#### 3. proof

##### img 1
![image](https://user-images.githubusercontent.com/95204402/175454616-8411e733-3d68-410e-8551-fbc05df65085.png)

##### img 2
![image](https://user-images.githubusercontent.com/95204402/175455965-fc377d1d-a395-4fc5-8ded-78e1cde607a7.png)

#### 4. references
- https://chrome.google.com/webstore/detail/custom-javascript-for-web/ddbjnfjiigjmcpcpkmhogomapikjbjdk - Custom JavaScript for Websites 2
- https://chrome.google.com/webstore/detail/page-refresh/hmooaemjmediafeacjplpbpenjnpcneg - Page Refresh - add 5 seconds

#### 5. concept "no-code"
Page Refresh+Custom JavaScript for Websites 2+code = remover follower

**Reply by rilysh** [reply]
They mean "a correct way, and a specific person", this will literally going to remove all followers.

**Comment by Raymond-Ly** [comment]
The trick is to block the person and unblock.

**Reply by adriano66** [reply]
Welp, at least we can do this, thanks for the hint. There should be a button in the UI that does exactly that if the result is the same

**Comment by jfullstackdev** [comment]
yup, just wondering too, if that is just a spam user, then blocking that user makes more sense, that's the  workaround by now

we don't know whether the requested feature here is included in their roadmap

**Comment by vsg24** [comment]
+1 for this. I received a wave of (what I suspect to be) fake/spam users following me, of course I can block them but it would still be easier to simply see your followers list and remove them on the spot

**Comment by ChidiChuks** [comment]
That will be a great feature to have

**Comment by shivangshah2000** [comment]
+1 to this. This is a must-have feature.

**Comment by Subilan** [comment]
+1 for this. I've encountered so many automatic follows done by random people for whatever reasons. They usually have a few thousand followers and followings. One guy even got **200k+** following in his profile 😕 . This kind of follow makes no sense to me and I wish to remove them from my follower list.

Block & Unblock is a workaround for now.

**Reply by dreua** [reply]
:100: agreed. I even reported a few of those because in my view that kind of automatic following is spam/abuse. Unfortunately Github doesn't care at all. One more reason to move on to other platforms, Codeberg being one example.

**Reply by An0n-00** [reply]
yep totally agree💯

like i already commented on this issue on [May 15, 2024](https://github.com/orgs/community/discussions/18941#discussioncomment-9447851) and still nothing new😐.... @github? any update.

(and i can also actually recommend codeberg. github is like instagram/social media. only because everyone is on github you also have to have github.....)

**Comment by erunaraina** [comment]
+1 - would actually prefer a setting to disable Followers entirely.
